src/visu.f90

```diff
@@ -457,6 +460,8 @@ subroutine write_field(f1, pathname, filename, num, skip_ibm)
     character(len=*), intent(in) :: pathname, filename, num 
     logical, optional, intent(in) :: skip_ibm
 
+    real(mytype), dimension(xsize(1),xsize(2),xsize(3)) :: local_array
```

Comment:
Can we avoid a local allocation here, e.g. using ta1?

Reply:
I was using di1 first, but some part of the BC-* functions are sending the array di1 already. Avoiding a local array would require to overwrite the input.

I am not sure about the behavior if the input `f1` is the array `ta1`, declared as intent(in), and then we use the variable `ta1` inside the subroutine and modify it...

Reply:
Or we can add a 3D array of size xsize(1),xsize(2),xsize(3) inside the visu module, allocate it at the start, deallocate it at the end.